sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptionsImpl.java

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
```

Comment:
should this class in implementation package?

Reply:
No - just creates way more to-be-bridged cases. And it is not a typical implementation - I just had to use this hack to avoid inheriting CosmosQueryRequestOptions

Reply:
But I will make it an internal class to ComsosQueryRequestOptions - that should address this comment and also ensures the code that needs to be modified when there are any new properties to query request options (that should not also be available for readManyRequestOptions) is shared